opm/simulators/wells/WellBhpThpCalculator.cpp

```diff
@@ -839,6 +866,114 @@ bruteForceBracket(const std::function<double(const double)>& eq,
+                 const Well& well,
+                 const std::vector<double>& rates,
+                 const SummaryState& summaryState,
+                 DeferredLogger& deferred_logger) const
+{
+    assert(int(rates.size()) == 3); // the vfp related only supports three phases now.
+    assert(well_.isProducer()); // only valid for producers 
+
+    static constexpr int Gas = BlackoilPhases::Vapour;
+    static constexpr int Oil = BlackoilPhases::Liquid;
+    static constexpr int Water = BlackoilPhases::Aqua;
+
+    const double aqua = rates[Water];
+    const double liquid = rates[Oil];
+    const double vapour = rates[Gas];
+    const double thp = well_.getTHPConstraint(summaryState);
+
+    const auto& controls = well.productionControls(summaryState);
+    const auto& wfr =  well_.vfpProperties()->getExplicitWFR(controls.vfp_table_number, well_.indexOfWell());
+    const auto& gfr = well_.vfpProperties()->getExplicitGFR(controls.vfp_table_number, well_.indexOfWell());
+
+    const auto& table = well_.vfpProperties()->getProd()->getTable(controls.vfp_table_number);
+    const bool use_vfpexplicit = well_.useVfpExplicit();
+
+    detail::VFPEvaluation bhp = detail::bhp(table, aqua, liquid, vapour, thp, well_.getALQ(well_state), wfr, gfr, use_vfpexplicit);
+
+    bhp.value = bhp.value + getVfpBhpAdjustment(bhp.value, thp);
+
+    if (bhp.dflo >= 0) {
+        return true;
+    } else {    // maybe check if ipr is available
+        const auto ipr = getFloIPR(well_state, well, summaryState);
+        return bhp.dflo + 1/ipr.second >= 0; 
+    }                  
+}
+
+std::optional<double> WellBhpThpCalculator::
+estimateStableBhp(const WellState& well_state,
+                  const Well& well,
+                  const std::vector<double>& rates,
+                  const double rho,
+                  const SummaryState& summaryState,
+                  DeferredLogger& deferred_logger) const
+{   
+    // Given a *converged* well_state with ipr, estimate bhp of the stable solution 
+    const auto& controls = well.productionControls(summaryState);
+    const double thp = well_.getTHPConstraint(summaryState);
+    const auto& table = well_.vfpProperties()->getProd()->getTable(controls.vfp_table_number);
+
+    const double aqua = rates[BlackoilPhases::Aqua];
+    const double liquid = rates[BlackoilPhases::Liquid];
+    const double vapour = rates[BlackoilPhases::Vapour];
+    double flo = detail::getFlo(table, aqua, liquid, vapour);
+    double wfr, gfr;
+    if (well_.useVfpExplicit() || -flo < table.getFloAxis().front()) {
+        wfr =  well_.vfpProperties()->getExplicitWFR(controls.vfp_table_number, well_.indexOfWell());
+        gfr = well_.vfpProperties()->getExplicitGFR(controls.vfp_table_number, well_.indexOfWell());
+    } else {
+        wfr = detail::getWFR(table, aqua, liquid, vapour);  
+        gfr = detail::getGFR(table, aqua, liquid, vapour);   
+    }
+    if (wfr <= 0.0 && gfr <= 0.0) {
+        // warning message
```

Comment:
Det er tanken som teller...

Reply:
Yes :). In general, we should not end up using zero fractions in the interpolation. There may be cases, however, where it's indendent, so just removed the mute `// warning`